rd-cpp/CMakeLists.txt

```diff
@@ -29,7 +29,10 @@ if (CMAKE_COMPILER_IS_GNUCC OR CMAKE_COMPILER_IS_GNUCXX OR CMAKE_CXX_COMPILER_ID
         set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -Wa,-mbig-obj")
     endif ()
 elseif (MSVC)
-    set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS}")
+    set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} /bigobj")
+    add_definitions(-D_CRT_SECURE_NO_WARNINGS)
+    add_compile_options(/wd4996     # POSIX name is deprecated
+                        /wd4250)    # method inheritance via dominance
```

Comment:
Same goes for compile options, would be better to have `target_compile_options` specified for problematic targets and avoid disabling warnings project wide